### PR TITLE
Add Elastic Solutions tracks

### PR DESCRIPTION
**Pull request by DJRickyB**
This PR introduces solutions-oriented tracks for Logs and Security, including tests and documentation.

Tested with
```bash
   esrally list tracks --track-repository djrickyb --track-revision elastic
   esrally info --track-repository djrickyb --track-revision elastic --track elastic/security
   esrally info --track-repository djrickyb --track-revision elastic --track elastic/logs
```

### CODE REVIEW

File: elastic/logs/README.md [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+	
+2. How much data will be sent to my cluster during indexing?
+	
+	Similar to data generation, the associated expansion of converting raw messages to JSON means that more data will be sent to Elasticsearch then the value of `raw_data_volume_per_day` * `(end_date - start_date)`. We estimate this expansion factor to be around 10x based on the default corpus ratios. This value is highly variable across different corpora - if you modify the corpus ratios the expansion factor may therefore increase or decrease accordingly.
+	
+3. Why is more data generated and sent than specified by `raw_data_volume_per_day` * `(end_date - start_date)`? 
+
+	See question (2)
+
+4. Can data be indexed more than once? Do i need to think about duplication?
```
Comment (b-deam) [inline elastic/logs/README.md:414]: Nit:
```suggestion
4. Can data be indexed more than once? Do I need to think about duplication?
```

File: elastic/logs/README.md [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+	
+	- Users can limit the size of the generated file using the parameter `max_generated_corpus_size`, in order to conserve disk space. Should this value be less than the required data needing indexing, the track will reuse the generated data - looping over it and inturn causing data duplication in Elasticsearch. The duplication caused by this behaviour converges to 0 as `max_generated_corpus_size` approaches approximately `raw_data_volume_per_day * (end_date - start_date) * 10` (assuming default corpus ratios).
+	
+5. How do i create a scenario where data is bulk loaded, and once sufficient volume is available, before querying and indexing occurs concurrently?
```
Comment (b-deam) [inline elastic/logs/README.md:424]: Nit:
```suggestion
5. How do I create a scenario where data is bulk loaded, and once sufficient volume is available, before querying and indexing occurs concurrently?
```

File: elastic/logs/slm/all.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  "repository": "many-shards",
+  "config": {
+    "indices": ["*"]
+  },
+  "retention": {
+    "expire_after": "1d",
+    "min_count": 5,
+    "max_count": 10
+  }
+}
```
Comment (b-deam) [inline elastic/logs/slm/all.json:13]: Nit: missing EOF newline

File: elastic/logs/tasks/field-caps.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "field-caps",
+  "operation": {
+     "operation-type": "field-caps",
+     "index": "auditbeat*"
+  },
+  "iterations": 200,
+  "warmup-iterations": 10,
+  "clients": 2
+}
```
Comment (b-deam) [inline elastic/logs/tasks/field-caps.json:10]: Nit: missing EOF newline

File: elastic/logs/tasks/many-shards-repo-setup.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  "name": "register-snapshot-repository",
+  "operation": {
+    "operation-type": "create-snapshot-repository",
+    "repository": "many-shards",
+    "body": {
+      "type": {{ p_snapshot_repo_type | tojson }},
+      "settings": {{ p_snapshot_repo_settings | tojson(indent=2)}}
+    }
+  }
+}
```
Comment (b-deam) [inline elastic/logs/tasks/many-shards-repo-setup.json:12]: Nit: missing EOF newline

File: elastic/logs/tasks/many-shards-setup.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+},
+{
+  "name": "delete-all-auditbeat-datastreams",
+  "tags": ["setup"],
+  "operation": {
+    "operation-type": "delete-data-stream",
+    "data-stream": "auditbeat*"
+  }
+},
+{% include "tasks/index-setup.json" %}
```
Comment (b-deam) [inline elastic/logs/tasks/many-shards-setup.json:23]: Nit: missing EOF newline

File: elastic/logs/tasks/slm-setup.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "insert-slm",
+  "tags": ["setup"],
+  "operation": {
+    "operation-type": "create-slm",
+    "param-source": "add-track-path"
+  }
+}
```
Comment (b-deam) [inline elastic/logs/tasks/slm-setup.json:8]: Nit: missing EOF newline

**Review by michaelbaamonde — APPROVED** [review]
Worked for me locally. I left a nitpick about the `.gitignore`, but not a blocker.

File: elastic/.gitignore [hunk 8]
```diff
@@ -0,0 +1,6 @@
+NOTES.md
```
[the remark this answers is not in the document]
Comment (DJRickyB) [reply elastic/.gitignore]: i've taken this out and only additionally added `.venv` to the top-level `.gitignore`. That was what was necessary after a `make install && make test`

**Review by b-deam — APPROVED** [review]
Thanks for putting up with my nits.

Tested locally and works, LGTM!

**Pull request merged**